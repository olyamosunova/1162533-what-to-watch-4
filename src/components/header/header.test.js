import React from 'react';
import renderer from 'react-test-renderer';

import Header from './header.jsx';

const userData = {
  id: 1,
  email: `sadas@dsasd.ru`,
  name: `asdasd`,
  avatarUrl: `https://4.react.pages.academy/wtw/asda.jpg`,
};

describe(`Header`, () => {
  it(`Should render correctly when is main page and user signed in`, () => {
    const tree = renderer
      .create(<Header
        authorizationStatus={``}
        userData={userData}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render correctly when is main page and user is not signed in`, () => {
    const tree = renderer
      .create(<Header
        authorizationStatus={``}
        userData={userData}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
