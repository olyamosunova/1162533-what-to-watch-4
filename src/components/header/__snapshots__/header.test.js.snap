// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render correctly when is main page and user is not signed in 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="main.html"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  />
</header>
`;

exports[`Header Should render correctly when is main page and user signed in 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="main.html"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  />
</header>
`;
